src/Switch-Arduino_MCP4728/main.py

```diff
@@ -0,0 +1,269 @@
+    """
+
+    def __init__(self) -> None:
+        EmptyDevice.__init__(self)
+
+        self.shortname = "Arduino MCP4728"
+        self.variables = []
+        self.units = []
+
+        self.port_manager = True
+        self.port_types = ["COM"]
+        self.port_properties = {
+            "EOL": "\n",
+            "timeout": 15,
+            "baudrate": 115200,
+        }
+
+        self.multi_pins = False
+        self.multi_mcp = False
+        self.max_voltage = None
+        self.addresses = []
+
+        self.unit = {
+            "Voltage in V": "V",
+            "Output in %": "%",
+        }
+
+    def set_GUIparameter(self):
+        return {
+            "Channel": ["0", "1", "2", "3", "all"],
+            "I2C Address": "0",
+            "SweepMode": ["Voltage in V", "Output in %"],
+            "Voltage reference": ["External", "Internal 2.048 V", "Internal 4.096 V"],
+            "External voltage in V": 5.0,
+        }
+
+    def get_GUIparameter(self, parameter={}):
+        # Handle output pins
+        channel = parameter["Channel"]
+        if channel == "all":
+            self.multi_pins = True
+        else:
+            self.pin = int(channel)
+
+        # Handle I2C addresses
+        self.addresses = []
+        address_list = parameter["I2C Address"].split(",")
+        if len(address_list) == 1:
+            self.addresses.append(int(parameter["I2C Address"]))
+        else:
+            for addr in address_list:
+                self.addresses.append(int(addr))
+            self.multi_mcp = True
+
+        self.sweepmode = str(parameter["SweepMode"])
+
+        # Reference can either be internal (2.048 V or 4.096 V) or external (custom V)
+        self.reference_voltage = str(parameter["Voltage reference"])
+        if self.reference_voltage == "External":
+            self.external_voltage = float(parameter["External voltage in V"])
+
+        # Set variables and units
+        if channel == "all" and self.multi_mcp:
+            channel_num = len(self.addresses) * 4
+        elif channel == "all" and not self.multi_mcp:
+            channel_num = 4
+        else:
+            channel_num = 1
+
+        self.variables = []
+        self.units = []
+        for n in range(channel_num):
+            # If single channel, use the correct pin number
+            if channel_num == 1:
+                n = self.pin
+            self.variables.append(f"Channel{n}")
+            self.units.append(self.unit[self.sweepmode])
+
+    """ here, semantic standard functions start that are called by SweepMe! during a measurement """
+
+    def connect(self):
+        # Set Name/Number of COM Port as key
+        instance_key = f"mcp4728_{self.port.port.port}"
+
+        if instance_key not in self.device_communication:
+            # Wait for Arduino initialization
+            self.port.read()
+            self.device_communication[instance_key] = "Connected"
+
+    def configure(self):
```

Comment:
The driver also needs an 'unconfigure' function. Otherwise, all channels stay at their last value. Here, I would set all channels back to 0.